test/observe_test_utils.dart

```diff
@@ -7,8 +7,27 @@ library observe.test.observe_test_utils;
 import 'dart:async';
 import 'package:observe/observe.dart';
 import 'package:observe/mirrors_used.dart' as mu; // to make tests smaller
-import 'package:unittest/unittest.dart';
+import 'package:observe/src/dirty_check.dart';
+import 'package:test/test.dart' hide test, group, setUp, tearDown;
+import 'package:test/test.dart' as original_test
+    show test, group, setUp, tearDown;
+
 export 'package:observe/src/dirty_check.dart' show dirtyCheckZone;
+export 'package:test/test.dart' hide test, group, setUp, tearDown;
+
+/// Custom implementations of the functions from `package:test`. These ensure
+/// that the body of all test function are run in the dirty checking zone.
+test(String description, body()) => original_test.test(
+    description, () => dirtyCheckZone().bindCallback(body)());
```

Comment:
> I suppose we could bind test callbacks to the zone they're declared in, but if the zone is stateful then you're setting yourself up for hidden state dependencies between tests.

Good point, although I don't think it would be any worse than the solution implemented here in that regard.

> Another possibility would be to provide a surround() function that would work like a combination of setUp() and tearDown()

This sounds like it would help, but I don't know if its worth investing time in or not.